plugins/progress/progress.plugin.sh

```diff
@@ -29,32 +31,57 @@ function delay()
-  if [ $CURRENT_PROGRESS -le 30 -a $PARAM_PROGRESS -ge 30 ]; then printf "[######....................] (30%) %s \r" "$PARAM_PHASE" ; delay; fi;
-  if [ $CURRENT_PROGRESS -le 35 -a $PARAM_PROGRESS -ge 35 ]; then printf "[#######...................] (35%) %s \r" "$PARAM_PHASE" ; delay; fi;
-  if [ $CURRENT_PROGRESS -le 40 -a $PARAM_PROGRESS -ge 40 ]; then printf "[########..................] (40%) %s \r" "$PARAM_PHASE" ; delay; fi;
-  if [ $CURRENT_PROGRESS -le 45 -a $PARAM_PROGRESS -ge 45 ]; then printf "[#########.................] (45%) %s \r" "$PARAM_PHASE" ; delay; fi;
-  if [ $CURRENT_PROGRESS -le 50 -a $PARAM_PROGRESS -ge 50 ]; then printf "[##########................] (50%) %s \r" "$PARAM_PHASE" ; delay; fi;
-  if [ $CURRENT_PROGRESS -le 55 -a $PARAM_PROGRESS -ge 55 ]; then printf "[###########...............] (55%) %s \r" "$PARAM_PHASE" ; delay; fi;
-  if [ $CURRENT_PROGRESS -le 60 -a $PARAM_PROGRESS -ge 60 ]; then printf "[############..............] (60%) %s \r" "$PARAM_PHASE" ; delay; fi;
-  if [ $CURRENT_PROGRESS -le 65 -a $PARAM_PROGRESS -ge 65 ]; then printf "[#############.............] (65%) %s \r" "$PARAM_PHASE" ; delay; fi;
-  if [ $CURRENT_PROGRESS -le 70 -a $PARAM_PROGRESS -ge 70 ]; then printf "[###############...........] (70%) %s \r" "$PARAM_PHASE" ; delay; fi;
-  if [ $CURRENT_PROGRESS -le 75 -a $PARAM_PROGRESS -ge 75 ]; then printf "[#################.........] (75%) %s \r" "$PARAM_PHASE" ; delay; fi;
-  if [ $CURRENT_PROGRESS -le 80 -a $PARAM_PROGRESS -ge 80 ]; then printf "[####################......] (80%) %s \r" "$PARAM_PHASE" ; delay; fi;
-  if [ $CURRENT_PROGRESS -le 85 -a $PARAM_PROGRESS -ge 85 ]; then printf "[#######################...] (90%) %s \r" "$PARAM_PHASE" ; delay; fi;
-  if [ $CURRENT_PROGRESS -le 90 -a $PARAM_PROGRESS -ge 90 ]; then printf "[##########################] (100%) %s \r" "$PARAM_PHASE"; delay; fi;
-  if [ $CURRENT_PROGRESS -le 100 -a $PARAM_PROGRESS -ge 100 ];then printf 'Done!                                            \n' ; delay; fi;
-
-  CURRENT_PROGRESS=$PARAM_PROGRESS;
+    local value=$1;
+    local message=$2;
+
+    if [ -z $value ]; then
+      printf "Usage: progress <value> [message]\n\n"
+      printf "Options:\n"
+      printf "  value    The value for the progress bar. Use 0 to reset.\n"
+      printf "  message  The optional message to display next to the progress bar.\n"
+      return 0
+    fi
+
+    if [ $value -lt 0 ]; then
+      _omb_log_error "invalid value: value' (expect: 0-100)" >&2
+      return 2
+    fi
+
+    # Reset the progress value
+    if [ $value -eq 0 ]; then
+      _omb_plugin_progress_value=0;
+      return 0
+    fi
+
+    # Clear the message whitespace
+    local size=40-${#message}
+    local whitespace=""
+    for ((i=0; i<size; i++)); do
+      whitespace="$whitespace "
+    done
+
+    if [ $_omb_plugin_progress_value -le 0 -a $value -ge 0 ]  ; then echo -ne "[--------------------------] (0%)  $message$whitespace \r"  ; delay; fi;
```

Comment:
What is the purpose of the padding by `$whitespace`?

Reply:
To clear out any previous message text so it doesn't overlap.

If you have Hello World on one progress. And then Yo on the second, the message ends up being "Yollo World"

Reply:
OK, in this case we usually put EL [`tput el` (terminfo), `tput ce` (termcap), or `\e[K` (ANSI)].

Reply:
What I did is absolutely gross.Mind having a look? lol

Reply:
I left comments in new conversations.